pycheribuild/projects/cross/apache.py

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright (c) 2021 SRI International
+# All rights reserved.
+#
+# This software was developed by SRI International and the University of
+# Cambridge Computer Laboratory (Department of Computer Science and
+# Technology) under DARPA contract HR0011-18-C-0016 ("ECATS"), as part of the
+# DARPA SSITH research programme.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+#
+from .crosscompileproject import (CrossCompileAutotoolsProject, DefaultInstallDir, GitRepository, SubversionRepository)
+
+
+class BuildPcre(CrossCompileAutotoolsProject):
+    target = "pcre"
+    project_name = "pcre"
+
+    repository = SubversionRepository("svn://vcs.pcre.org/pcre",
+                                      default_branch="code/trunk")
+
+    native_install_dir = DefaultInstallDir.BOOTSTRAP_TOOLS
+    cross_install_dir = DefaultInstallDir.ROOTFS_LOCALBASE
+
+    def configure(self, **kwargs):
+        self.run_cmd("autoreconf", "-i", cwd=self.source_dir)
+        super().configure(**kwargs)
+
+
+class BuildApr(CrossCompileAutotoolsProject):
+    target = "apr"
+    project_name = "apr"
+
+    repository = GitRepository("https://github.com/CTSRD-CHERI/apr.git",
+                               default_branch="cheri")
+
+    dependencies = ["libexpat"]
+
+    native_install_dir = DefaultInstallDir.BOOTSTRAP_TOOLS
+    cross_install_dir = DefaultInstallDir.ROOTFS_LOCALBASE
+
+    def configure(self, **kwargs):
+        self.configure_args.extend([
+            "--enable-debug",
```

Comment:
I added this (and a similar one for apache) to get debug symbols.

It doesn't look like there's a generic "enable debug mode" knob in cheribuild projects on which I would perhaps make this flag conditional?

Reply:
Ah, used 'build_type.is-debug'.